plugins/modules/purefa_user.py

```diff
@@ -103,116 +135,158 @@
+                    names=[module.params["name"]], timeout=ttl
+                )
+                if res.status_code != 200:
+                    module.fail_json(
+                        msg="Failed to create API token. Error: {0}".format(
+                            res.errors[0].message
                         )
-            except Exception:
-                module.fail_json(
-                    msg="Local User {0}: Creation failed".format(module.params["name"])
+                    )
+                api_token = list(res.items)[0].api_token.token
+            if module.params["public_key"]:
+                res = array.patch_admin(
+                    names=[module.params["name"]],
+                    admin=AdminPatch(public_key=module.params["public_key"]),
                 )
+                if res.status_code != 200:
+                    module.fail_json(
+                        msg="Failed to add SSH key. Error: {0}".format(
+                            res.errors[0].message
+                        )
+                    )
     else:
         if module.params["password"] and not module.params["old_password"]:
             module.exit_json(changed=changed)
-        if module.params["password"] and module.params["old_password"]:
-            if module.params["old_password"] and (
-                module.params["password"] != module.params["old_password"]
-            ):
-                passwd_changed = True
-                if not module.check_mode:
-                    try:
-                        array.set_admin(
-                            module.params["name"],
-                            password=module.params["password"],
-                            old_password=module.params["old_password"],
-                        )
-                    except Exception:
-                        module.fail_json(
-                            msg="Local User {0}: Password reset failed. "
-                            "Check old password.".format(module.params["name"])
+        if (
+            module.params["password"]
+            and module.params["old_password"]
+            and module.params["password"] != module.params["old_password"]
+        ):
+            passwd_changed = True
+            if not module.check_mode:
+                res = array.patch_admins(
+                    names=[module.params["name"]],
+                    admin=AdminPatch(
+                        password=module.params["password"],
+                        old_password=module.params["old_password"],
+                    ),
+                )
+                if res.status_code != 200:
+                    module.fail_json(
+                        msg="Local User {0} password reset failed. Error: {1}"
+                        "Check old password.".format(
+                            module.params["name"], res.errors[0].message
                         )
-            else:
+                    )
+        if module.params["api"]:
```

Comment:
We've not set api_changed = True in this block, hence ansible reports just ok even though change has happened.
Please check and add it in appropriate place.